### DISCUSSION TITLE
Regarding the matrix elements of the phonon velocity operator

### DISCUSSION THREAD

**Discussion by sartaz20028** [phoebe-team/phoebe#220] [Features]
I want to calculate the matrix element of the phonon velocity operator (v^α_qbb') through Phoebe code. In addition to this, can I also calculate off-diagonal terms of the matrix element of the phonon velocity in which real and imaginary parts are included? If yes, then please discuss the procedure.

**Comment by jcoulter12** [comment]
Hi, 

This should be possible, though it's not a default behavior of the code, as it could produce some fairly large files that most users would not find necessary. 

Would it be sufficient if you could run the phononBandsApp for the grid of your choice and the code outputs this to a json file? Something like this would not be too big of a modification, and I could probably add it in the next few days. 

Thanks, 
Jenny

**Comment by sartaz20028** [comment]
Thank you for replying.
Actually, I need the phonon group velocity matrix, particularly the interband contribution, which is an important concept when considering the coupling between different phonon modes (bands) in a crystal. However, when I run the phononBandsApp for the grid of our choice, it produces only phonon group velocity (refer to vg_normal.png). But I have need the interband phonon velocity matrix (refer to interband_vg.png). For your reference, I have attached the formula for both. 
![interband_vg](https://github.com/user-attachments/assets/f465cd69-9137-4207-a136-d2d1bfbfbd8e)
![vg_normal](https://github.com/user-attachments/assets/63cc2aec-ab19-4640-aca3-dea0341c216b)

**Comment by jcoulter12** [comment]
Hi, 

First, as the main developer of Phoebe, I do indeed know the importance of the phonon group velocities and also that they are not output by the phononBandsApp, as I wrote it. If you read my earlier message again, you can see I am considering if I have the time to add this feature for you. 

With respect to your request, the formulas you provided are not the right ones to use, and therefore are not implemented in Phoebe, as they do not work in the presence of degeneracies. See Sec IX in https://journals.aps.org/prx/abstract/10.1103/PhysRevX.12.041011 for the details about the correct expressions, which are implemented in Phoebe. 
 
It is likely that I will add the ability to print the real and imaginary parts of the velocity operator elements, but I am working on several updates at the moment, so I will add this feature as soon as my schedule permits. Ideally it will not be too long. 

Are you trying to use the Wigner transport formalism in the PRX linked above? 

Jenny

**Comment by sartaz20028** [comment]
Yes

**Reply by jcoulter12** [reply]
In that case, the Wigner transport correction is implemented in the code already. The conductivity with the Wigner contribution is simply output in a JSON file at the end of the phononTransportApp. In the Phonon Transport tutorial, you can see in the section describing the outputs, we mention: 

`rta_wigner_coefficients.json: contains the RTA transport coefficients + the Wigner transport correction.`

Which contains that correction already calculated.
